### PR TITLE
Add option to allow for host users not to be deleted

### PR DESCRIPTION
**Pull request by lxea**
This adds a new role option called create_host_users_mode which allows for it to be configured to not delete users when a session ends. The old `create_host_user` option will be deprecated with this and if the create_host_user_mode option is set it'll always be preffered over `create_host_user`.

If the deprecated option is set, the new mode option will default to dropping users as is the current behavior.

### CODE REVIEW

**Review by smallinsky — COMMENTED** [review]
First pass. Could you fix CI and  UT build. Also wonder if you can share any reference for this change ?

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -2447,6 +2447,13 @@ message RoleOptions {
     (gogoproto.jsontag) = "create_db_user",
     (gogoproto.customtype) = "BoolOption"
   ];
+
+  // CreateHostUserMode allows users to be automatically created on a
+  // host when not set to off
+  string CreateHostUserMode = 28 [
+    (gogoproto.jsontag) = "create_host_user_mode,omitempty",
+    (gogoproto.casttype) = "github.com/gravitational/teleport/api/constants.HostUserMode"
```
Comment (Joerger) [inline api/proto/teleport/legacy/types/types.proto:2455]: Can we use a protobuf enum instead here? Keep in mind we want to move away from gogoproto _eventually_, so the less we depend on it the better.

**Comment by r0mant** [comment]
@Joerger @gabrielcorado @smallinsky Can you guys review this?

**Review by smallinsky — APPROVED** [review]
Sorry I had stashed uncommitted review. 

Mostly LGT though  I would love to see UT coverage for Marshal/Unmarshal function.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -2325,6 +2325,19 @@ message RoleSpecV6 {
   ];
 }
 
+// CreateHostUserMode determines whether host user creation should be
+// disabled or if host users should be cleaned up or remain after
+// sessions end.
+enum CreateHostUserMode {
+  HOST_USER_MODE_UNDEFINED = 0;
```
Comment (Joerger) [inline api/proto/teleport/legacy/types/types.proto:2332]: nit: [Use "*_UNSPECIFIED"](https://protobuf.dev/programming-guides/dos-donts/#do-include-an-unspecified-value-in-an-enum)
```suggestion
  HOST_USER_MODE_UNSPECIFIED = 0;
```

File: api/proto/teleport/legacy/types/types.proto [hunk 3]
```diff
@@ -2325,6 +2325,19 @@ message RoleSpecV6 {
   ];
 }
 
+// CreateHostUserMode determines whether host user creation should be
+// disabled or if host users should be cleaned up or remain after
```
Comment (Joerger) [inline api/proto/teleport/legacy/types/types.proto:2329]: nit:
```suggestion
// disabled or if host users should be cleaned up or kept after
```

File: lib/services/role_test.go [hunk 4]
```diff
@@ -7221,11 +7253,131 @@ func TestHostUsers_CanCreateHostUser(t *testing.T) {
...
+			false,
+			types.CreateHostUserMode_HOST_USER_MODE_OFF,
+		),
+		createDefaultTCWithMode(
+			"test can create when create host user mode is drop",
+			true,
+			types.CreateHostUserMode_HOST_USER_MODE_DROP,
+		),
+		createDefaultTCWithMode(
+			"test can create when create host user mode is remain",
```
Comment (Joerger) [inline lib/services/role_test.go:7289]: nit:
```suggestion
			"test can create when create host user mode is keep",
```

**Review by Joerger — APPROVED** [review]
LGTM, just some nit-picks

**Pull request merged**